rbac/management/audit_log/model.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Model for audit logging."""
+from django.db import models
+from django.utils import timezone
+
+from api.models import Tenant, TenantAwareModel
+
+
+class AuditLog(TenantAwareModel):
+    """An audit log."""
+
+    GROUP = "group"
+    ROLE = "role"
+    USER = "user"
+    PERMISSION = "permission"
+    RESOURCE_CHOICES = (
+        (GROUP, "Group"),
+        (ROLE, "Role"),
+        (USER, "User"),
+        (PERMISSION, "Permission"),
+    )
+
+    DELETE = "delete"
+    ADD = "add"
+    EDIT = "edit"
+    CREATE = "create"
+    REMOVE = "remove"
+    ACTION_CHOICES = (
+        (DELETE, "Delete"),
+        (ADD, "Add"),
+        (EDIT, "Edit"),
+        (CREATE, "Create"),
+        (REMOVE, "Remove"),
+    )
+
+    created_at = models.DateTimeField(default=timezone.now)
+    requester = models.TextField(max_length=255, null=False)
```

Comment:
as I understand there will be name of user(principal) name how created a log entry ?

I would recommend to store relation and I would name it as principal

```
   principal = models.ForeignKey(....
```

 with this we can track principal exactly - in  principal table 

Does it make sense to you regarding the assignment in jira ticket?

Reply:
I was thinking we have that separate, since we have to track whether a principal is being removed/added to a group, so thats why I have the requester field there which is tracking the user who created the log entry.

Is that alright?

Reply:
Ohhhhh, do you mean to have the requester be related to the principal table so that we can track the requester within the principal table? 

So the change would be like:
`requester =models.ForeignKey(Principals..)`

Reply:
yes exactly!